x-pack/plugins/ml/public/application/components/model_snapshots/revert_model_snapshot_flyout/create_calendar.tsx

```diff
@@ -0,0 +1,283 @@
+import { Anomaly } from '../../../jobs/new_job/common/results_loader/results_loader';
+import { useCurrentEuiTheme } from '../../../components/color_range_legend';
+import { LineChartPoint } from '../../../jobs/new_job/common/chart_loader/chart_loader';
+
+export interface CalendarEvent {
+  start: moment.Moment | null;
+  end: moment.Moment | null;
+  description: string;
+}
+
+interface Props {
+  calendarEvents: CalendarEvent[];
+  setCalendarEvents: (calendars: CalendarEvent[]) => void;
+  minSelectableTimeStamp: number;
+  maxSelectableTimeStamp: number;
+  eventRateData: LineChartPoint[];
+  anomalies: Anomaly[];
+  chartReady: boolean;
+}
+
+export const CreateCalendar: FC<Props> = ({
+  calendarEvents,
+  setCalendarEvents,
+  minSelectableTimeStamp,
+  maxSelectableTimeStamp,
+  eventRateData,
+  anomalies,
+  chartReady,
+}) => {
+  const maxSelectableTimeMoment = moment(maxSelectableTimeStamp);
+  const minSelectableTimeMoment = moment(minSelectableTimeStamp);
+
+  const { euiTheme } = useCurrentEuiTheme();
+
+  function onBrushEnd({ x }: XYBrushArea) {
+    if (x && x.length === 2) {
+      const end = x[1] < minSelectableTimeStamp ? null : x[1];
+      if (end !== null) {
+        const start = x[0] < minSelectableTimeStamp ? minSelectableTimeStamp : x[0];
+
+        setCalendarEvents([
+          ...calendarEvents,
+          {
+            start: moment(start),
+            end: moment(end),
+            description: createDefaultEventDescription(calendarEvents.length + 1),
+          },
+        ]);
+      }
+    }
+  }
+
+  function setStartDate(start: moment.Moment | null, index: number) {
+    const event = calendarEvents[index];
+    if (event === undefined) {
+      setCalendarEvents([
+        ...calendarEvents,
+        { start, end: null, description: createDefaultEventDescription(index) },
+      ]);
+    } else {
+      event.start = start;
+      setCalendarEvents([...calendarEvents]);
+    }
+  }
+
+  function setEndDate(end: moment.Moment | null, index: number) {
+    const event = calendarEvents[index];
+    if (event === undefined) {
+      setCalendarEvents([
+        ...calendarEvents,
+        { start: null, end, description: createDefaultEventDescription(index) },
+      ]);
+    } else {
+      event.end = end;
+      setCalendarEvents([...calendarEvents]);
+    }
+  }
+
+  const setDescription = (description: string, index: number) => {
+    const event = calendarEvents[index];
+    if (event !== undefined) {
+      event.description = description;
+      setCalendarEvents([...calendarEvents]);
+    }
+  };
+
+  function removeCalendarEvent(index: number) {
+    if (calendarEvents[index] !== undefined) {
+      const ce = [...calendarEvents];
+      ce.splice(index, 1);
+      setCalendarEvents(ce);
+    }
+  }
+
+  return (
+    <>
+      <EuiSpacer size="l" />
+      <div>Select time range for calendar event.</div>
+      <EuiSpacer size="m" />
+      <Chart
```

Comment:
Is it worth masking the calendar chart for the time period before the snapshot timestamp, to indicate you can't have a calendar event before the time of the snapshot? I tried selecting an interval which included the orange anomaly near the start of the time range, and took a moment to realize it only makes sense for dates after the time at which the replay will begin.

![image](https://user-images.githubusercontent.com/7405507/84499645-e6833e00-acaa-11ea-8000-46587b08aef9.png)

Reply:
I tested this and without a legend saying what the new mask represents, I thought it might add confusion to a very simple chart.